Updating icons

Christonn93 opened the pull request:
# Work that have been done

1. Installed material icons [material-icons](https://www.npmjs.com/package/material-icons)
2. Changed all old icons whit the new
3. Changed class name in `_components.scss`
4. Added logo as favicon

## List of icons
[Material Icons](https://marella.me/material-icons/demo/) - The list of icons to be used. 

# UPDATE

As suggested from @pretzL. 

Its created a `Icon.svelte` file under `components/ui`. 

## Usage
`<Icon iconName="the name of the icon" />`
Eks: 
`<Icon iconName="bookmark" />` 

<del></del>
<del>## Usage</del>
<del>For adding new icons use the following:</del>
<del>`<span class="material-icons">THE NAME OF THE ICON GOES HERE</span>` </del>

<del>eks: 
<del>`<span class="material-icons">settings</span>` will display: ![image](https://github.com/Fermain/-mollify/assets/46559129/b0325eee-a777-4521-aa67-377103a30fd0)</del>

<del>To get the style correct, you will need to include `import 'material-icons/iconfont/material-icons.css';` in the script tag on the `page_name.svelte` where the icons will be used. 

</del>

## Style 
In the `_components.scss` file there is a `@mixin icon` section that was used to add style to icons before the change. 
I removed `.icon-f` and `.icon-o` and replaced the class name with `.material-icons` but did not see any change in the style. Not sure why this is. 

I also added the logo as a `favicon.ico`, For some reason it do not show up. Might work more on that.

Christonn93 commented on packages/lms/src/lib/components/header/Settings.svelte:
```diff
@@ -1,156 +1,160 @@
-			autohide: false
-		};
-		toastStore.trigger(userFeedback);
-	}
```
This have been changed du to formatting

Fermain commented on packages/lms/package.json:
```diff
@@ -18,6 +18,7 @@
   "dependencies": {
     "@floating-ui/dom": "^1.2.8",
     "@mollify/cli": "^0.2.11",
+    "@mollify/lms": "^0.2.11",
```
A project shouldn't install itself

Christonn93 replied:
This is not something I have added myself, After running all the installations according to the Readme. This is not a change I have done by my self.

Fermain replied:
I figured out what happened, you need to move the material install here and remove it from the root level package file - and remove this line.

Fermain commented on packages/lms/src/app.html:
```diff
@@ -2,12 +2,8 @@
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%sveltekit.assets%/favicon.png" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
```
This is intentional, it is overwritten by the build process. You may not have the asset in /static

Christonn93 replied:
Reverted the change

Fermain commented on packages/lms/src/routes/+layout.svelte:
```diff
@@ -15,6 +15,8 @@
   import EntityNav from '$lib/components/navigation/EntityNav.svelte';
   import { page } from '$app/stores';
 
+  import favicon from "../favicon.ico"
```
This is already handled

Christonn93 replied:
Reverted the change

Fermain reviewed (commented):
This PR goes a little bit outside of the scope of the task. There are no issues with favicon and if there are these should be reported and worked on in a separate issue

Christonn93 commented:
> This PR goes a little bit outside of the scope of the task. There are no issues with favicon and if there are these should be reported and worked on in a separate issue

Okey. Thanks for the heads up. Will keep this in mind. 
I have reverted the changes made when adding the favicon, removed the favicon from the file structure also.

Fermain commented on package.json:
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
-    "dotenv": "^16.0.3"
+    "dotenv": "^16.0.3",
+    "material-icons": "^1.13.11"
```
I think something went wrong with your install. This should be installed inside the lms project, not on the root. Please move this to the other package file.

Christonn93 replied:
Moving the material icons from `package.json` to `packages/lms/package.json`
Should the dotenv also be moved?

Christonn93 replied:
And should i remove this `"@mollify/lms": "^0.2.11",` from the `packages/lms/package.json` ?

Fermain replied:
Keep dotenv, and remove @mollify/lms

pretzL commented:
> To get the style correct, you will need to include import 'material-icons/iconfont/material-icons.css'; in the script tag on the page_name.svelte where the icons will be used.

Why require that developers jump through so many hops to add an icon to a page when we can create an `<Icon />` component instead? It can take in a `name` attribute which corresponds to the material-icon name, apply any default styling we wish and sets up all necessary imports.

Christonn93 commented:
> 

Sure. I can set this up. 
I'll make a update later on this later.

Christonn93 commented:
# UPDATE

As suggested from @pretzL. 

Its created a `Icon.svelte` file under `components/ui`. 

## Usage
`<Icon iconName="the name of the icon" />`
Eks: 
`<Icon iconName="bookmark" />`

pretzL reviewed (commented):
I'll review this once the formatting is in-line with the rest of the project. As-is it's impossible to see what you've actually changed.

Christonn93 commented:
> I'll review this once the formatting is in-line with the rest of the project. As-is it's impossible to see what you've actually changed.

Terminal output from prettier. 
![image](https://github.com/Fermain/-mollify/assets/46559129/411e6da1-0750-407b-93a2-dd0b90501126)

Would appreciate some guidance on how to resolve the formatting issue, since I do not find any issue with the formatting when its using the formatting settings that is provided  in the `.prettierrc` file and formatting the code based on that.

pretzL commented:
> Can it be that code have been pushed without farmatting?

Definitely

Christonn93 commented:
This is what my output in vcs is displaying: 

[output-vsc-prettier.txt](https://github.com/Fermain/-mollify/files/12577521/output-vsc-prettier.txt)

Fermain commented:
Prettier isn't running on component files at all

```
(base) oliver@Fermain -mollify % npm run format --workspace @mollify/lms  

> @mollify/lms@0.2.11 format
> prettier --plugin-search-dir . --write .

.eslintrc.cjs 29ms
.markdownlint.yaml 15ms
.prettierrc 7ms
package.json 3ms
playwright.config.ts 84ms
postcss.config.cjs 4ms
README.md 23ms
src/app.d.ts 3ms
src/app.html 31ms
src/app.postcss 21ms
src/index.test.ts 4ms
src/lib/stores/audio.ts 4ms
src/lib/stores/bookmarks.ts 2ms
src/lib/stores/files.ts 3ms
src/lib/styles/_components.scss 4ms
src/lib/styles/_iframe.scss 3ms
src/lib/styles/_markdown.scss 4ms
src/lib/styles/_permalink.scss 1ms
src/lib/styles/_toc.scss 2ms
src/lib/styles/styles.scss 2ms
src/lib/utils/bookmarking/AddRemove.ts 18ms
src/lib/utils/fuseSearch/flattenData.ts 7ms
src/lib/utils/fuseSearch/fuseSetup.ts 22ms
src/lib/utils/fuseSearch/generateRawSearchQuery.ts 13ms
src/lib/utils/fuseSearch/getSearchResults.ts 5ms
src/lib/utils/fuseSearch/parseRawSearchQuery.ts 7ms
src/lib/utils/fuseSearch/removeStopWords.ts 2ms
src/lib/utils/fuseSearch/stopWords.ts 6ms
src/lib/utils/fuseSearch/updateQueryString.ts 2ms
src/lib/utils/getCurrentPageEntityMeta.ts 4ms
src/lib/utils/getEntityChildren.ts 3ms
src/lib/utils/getEntityFrontmatter.ts 4ms
src/lib/utils/getEntityMeta.ts 3ms
src/lib/utils/getEntityMetaTree.ts 6ms
src/lib/utils/getEntitySlug.ts 2ms
src/lib/utils/sortChildrenByDependency.ts 5ms
src/lib/utils/storage/index.ts 1ms
src/lib/utils/storage/load.ts 2ms
src/lib/utils/storage/remove.ts 1ms
src/lib/utils/storage/save.ts 1ms
src/lib/utils/storage/update.ts 2ms
src/lib/utils/tts/createAudio.ts 2ms
src/lib/utils/tts/fetchAudio.ts 2ms
src/routes/+layout.server.ts 2ms
src/routes/+page.md 6ms
src/routes/api/getCurrentPage/+server.ts 3ms
src/routes/api/getEntityMetaTree/+server.ts 2ms
src/routes/api/molly/+server.ts 1ms
src/routes/api/search/+server.ts 2ms
src/routes/api/tags/+server.ts 6ms
src/routes/api/tts/createAudio/+server.ts 3ms
src/routes/api/tts/getAudio/+server.ts 2ms
src/routes/content/feu1/+page.md 3ms
src/routes/content/feu1/agency-1/+page.md 11ms
src/routes/content/feu1/agency-1/delivery/+page.md 2ms
src/routes/content/feu1/agency-1/discovery/+page.md 30ms
src/routes/content/feu1/agency-1/planning/+page.md 3ms
src/routes/content/feu1/agency-1/retrospective/+page.md 4ms
src/routes/content/feu1/agency-1/sprint/+page.md 2ms
src/routes/content/feu1/design/+page.md 11ms
src/routes/content/feu1/design/ca/+page.md 14ms
src/routes/content/feu1/design/ca/marking-criteria/+page.md 3ms
src/routes/content/feu1/design/module-1/+page.md 4ms
src/routes/content/feu1/design/module-1/content-management/+page.md 103ms
src/routes/content/feu1/design/module-1/fundamentals-layout/+page.md 52ms
src/routes/content/feu1/design/module-1/intro-branding/+page.md 46ms
src/routes/content/feu1/design/module-1/visual-hierarchy/+page.md 59ms
src/routes/content/feu1/design/module-2/+page.md 5ms
src/routes/content/feu1/design/module-2/colors-of-the-web/+page.md 36ms
src/routes/content/feu1/design/module-2/font-pairings/+page.md 41ms
src/routes/content/feu1/design/module-2/fundamentals-color/+page.md 64ms
src/routes/content/feu1/design/module-2/typography-basics/+page.md 75ms
src/routes/content/feu1/design/module-3/+page.md 2ms
src/routes/content/feu1/design/module-3/intro-ux/+page.md 26ms
src/routes/content/feu1/design/module-3/principles-ui-design/+page.md 29ms
src/routes/content/feu1/design/module-3/principles-ux-design/+page.md 30ms
src/routes/content/feu1/design/module-3/user-interface/+page.md 31ms
src/routes/content/feu1/design/module-4/+page.md 2ms
src/routes/content/feu1/design/module-4/creating-prototypes/+page.md 30ms
src/routes/content/feu1/design/module-4/creating-wireframes/+page.md 31ms
src/routes/content/feu1/design/module-4/intro-prototyping/+page.md 27ms
src/routes/content/feu1/design/module-4/intro-wireframes/+page.md 22ms
src/routes/content/feu1/html-css/+page.md 6ms
src/routes/content/feu1/html-css/ca/+page.md 10ms
src/routes/content/feu1/html-css/ca/marking-criteria/+page.md 1ms
src/routes/content/feu1/html-css/module-1/+page.md 7ms
src/routes/content/feu1/html-css/module-1/file-management/+page.md 25ms
src/routes/content/feu1/html-css/module-1/fonts/+page.md 27ms
src/routes/content/feu1/html-css/module-1/intro-css/+page.md 64ms
src/routes/content/feu1/html-css/module-1/intro-html/+page.md 46ms
src/routes/content/feu1/html-css/module-1/module-1-checkpoint/+page.md 11ms
src/routes/content/feu1/html-css/module-2/+page.md 4ms
src/routes/content/feu1/html-css/module-2/css-box-model/+page.md 20ms
src/routes/content/feu1/html-css/module-2/forms/+page.md 36ms
src/routes/content/feu1/html-css/module-2/images-icons/+page.md 29ms
src/routes/content/feu1/html-css/module-2/module-2-checkpoint/+page.md 8ms
src/routes/content/feu1/html-css/module-2/navigation/+page.md 14ms
src/routes/content/feu1/html-css/module-3/+page.md 5ms
src/routes/content/feu1/html-css/module-3/css-grids/+page.md 18ms
src/routes/content/feu1/html-css/module-3/flexbox/+page.md 29ms
src/routes/content/feu1/html-css/module-3/media-queries/+page.md 24ms
src/routes/content/feu1/html-css/module-3/module-3-checkpoint/+page.md 9ms
src/routes/content/feu1/html-css/module-3/positioning/+page.md 38ms
src/routes/content/feu1/html-css/module-4/+page.md 5ms
src/routes/content/feu1/html-css/module-4/animations-transitions/+page.md 18ms
src/routes/content/feu1/html-css/module-4/bug-fixing/+page.md 25ms
src/routes/content/feu1/html-css/module-4/common-layouts/+page.md 8ms
src/routes/content/feu1/html-css/module-4/css-variables/+page.md 15ms
src/routes/content/feu1/html-css/module-4/module-4-checkpoint/+page.md 6ms
src/routes/content/feu1/introduction/+page.md 9ms
src/routes/content/feu1/introduction/ca/+page.md 4ms
src/routes/content/feu1/introduction/module-1/+page.md 13ms
src/routes/content/feu1/introduction/module-1/file-delivery-lanekassen/+page.md 21ms
src/routes/content/feu1/introduction/module-1/moodle-reflective-journal/+page.md 25ms
src/routes/content/feu1/introduction/module-1/plagiarism-referencing/+page.md 19ms
src/routes/content/feu1/introduction/module-1/planning-communication/+page.md 29ms
src/routes/content/feu1/introduction/module-1/what-to-expect/+page.md 3ms
src/routes/content/feu1/javascript-1/+page.md 6ms
src/routes/content/feu1/javascript-1/ca/+page.md 11ms
src/routes/content/feu1/javascript-1/ca/marking-criteria/+page.md 2ms
src/routes/content/feu1/javascript-1/module-1/+page.md 3ms
src/routes/content/feu1/javascript-1/module-1/html-from-objects/+page.md 15ms
src/routes/content/feu1/javascript-1/module-1/html-in-functions/+page.md 13ms
src/routes/content/feu1/javascript-1/module-1/intro-concepts/+page.md 18ms
src/routes/content/feu1/javascript-1/module-1/manipulating-dom/+page.md 17ms
src/routes/content/feu1/javascript-1/module-2/+page.md 2ms
src/routes/content/feu1/javascript-1/module-2/callback-functions/+page.md 11ms
src/routes/content/feu1/javascript-1/module-2/dom-events/+page.md 9ms
src/routes/content/feu1/javascript-1/module-2/length-trim-eventlistener/+page.md 5ms
src/routes/content/feu1/javascript-1/module-2/more-dom-concepts/+page.md 7ms
src/routes/content/feu1/javascript-1/module-3/+page.md 3ms
src/routes/content/feu1/javascript-1/module-3/arrow-functions-promises/+page.md 19ms
src/routes/content/feu1/javascript-1/module-3/handling-errors/+page.md 7ms
src/routes/content/feu1/javascript-1/module-3/more-about-apis/+page.md 25ms
src/routes/content/feu1/javascript-1/module-3/query-string-parameters/+page.md 6ms
src/routes/content/feu1/javascript-1/module-4/+page.md 5ms
src/routes/content/feu1/javascript-1/module-4/form-validation/+page.md 10ms
src/routes/content/feu1/javascript-1/module-4/get-requests/+page.md 10ms
src/routes/content/feu1/javascript-1/module-4/single-api-results/+page.md 8ms
src/routes/content/feu1/javascript-1/module-4/skipping-certain-api-results/+page.md 6ms
src/routes/content/feu1/porfolio-1/+page.md 2ms
src/routes/content/feu1/project-exam-1/+page.md 2ms
src/routes/content/feu1/project-methodology/+page.md 5ms
src/routes/content/feu1/project-methodology/ca/+page.md 8ms
src/routes/content/feu1/project-methodology/ca/marking-criteria/+page.md 1ms
src/routes/content/feu1/project-methodology/module-1/+page.md 3ms
src/routes/content/feu1/project-methodology/module-1/agile-scrum/+page.md 28ms
src/routes/content/feu1/project-methodology/module-1/gantt/+page.md 29ms
src/routes/content/feu1/project-methodology/module-1/kanban-user-stories/+page.md 25ms
src/routes/content/feu1/project-methodology/module-1/sdlc/+page.md 24ms
src/routes/content/feu1/project-methodology/module-1/trello/+page.md 10ms
src/routes/content/feu1/semester-project-1/+page.md 12ms
src/routes/content/feu1/semester-project-1/marking-criteria/+page.md 5ms
src/routes/test/+page.md 7ms
svelte.config.js 4ms
tailwind.config.cjs 2ms
tsconfig.json 1ms
vite.config.ts 3ms
```

Christonn93 commented:
Is there something I) need to install after the change in the package.json? 

Trying to run formatting but getting this error
![image](https://github.com/Fermain/-mollify/assets/46559129/0b47eb4f-f48f-4daf-8f75-5779584976b3)
@Fermain

Fermain commented:
@Christonn93 I did it on my side and pushed, but I am super confused about whats going on with formatting at the moment.

Fermain commented:
@pretzL @Christonn93 I fixed the formatting by resolving the conflicts. A couple of icon changes fell out during this process, like Entity.svelte.

Christonn93 commented:
I will have a look at it after dinner, @Fermain

Christonn93 commented:
Still reciving this error: 
![image](https://github.com/Fermain/-mollify/assets/46559129/f2ad1b8f-7568-479e-859d-6d33d224f00c)

But icons in `Reader.svelte` and `Entity.svelte` is changed.

Fermain commented:
> Still reciving this error: ![image](https://user-images.githubusercontent.com/46559129/268037702-f2ad1b8f-7568-479e-859d-6d33d224f00c.png)
> 
> But icons in `Reader.svelte` and `Entity.svelte` is changed.

I assume that you have tried `npm i` from the top level of the repo?

pretzL commented on packages/lms/src/lib/components/header/Header.svelte:
```diff
@@ -13,7 +14,7 @@
     <div class="flex items-center">
       <button class="lg:hidden btn btn-sm mr-4" on:click={drawerOpen}>
-        <i class="icon-f">menu</i>
+        <Icon iconName="menu" />
```
```suggestion
        <Icon name="menu" />
```
There's no reason not to use the shorter `name` property since `name` in this case isn't a reserved keyword.

Fermain replied:
I agree.

pretzL commented on packages/lms/src/lib/components/header/Settings.svelte:
```diff
@@ -129,7 +130,7 @@
 
 <div>
-  <button class="btn hover:bg-primary-hover-token" use:popup={settings}> <i class="icon-f">settings</i></button>
+  <button class="btn hover:bg-primary-hover-token" use:popup={settings}> <Icon iconName="settings" /></button>
```
```suggestion
  <button class="btn hover:bg-primary-hover-token" use:popup={settings}> <Icon name="settings" /></button>
```

pretzL commented on packages/lms/src/lib/components/header/Settings.svelte:
```diff
@@ -141,14 +142,14 @@
-        <button class="btn hover:bg-primary-hover-token p-1" on:click={increaseFontSize}
-          ><i class="icon-f">text_increase</i></button
+        <button class="btn hover:bg-primary-hover-token p-1" on:click={increaseFontSize}>
+          <Icon iconName="text_increase" /></button
```
```suggestion
          <Icon name="text_increase" /></button
```

pretzL commented on packages/lms/src/lib/components/header/Settings.svelte:
```diff
@@ -141,14 +142,14 @@
-        <button class="btn hover:bg-primary-hover-token p-1" on:click={decreaseFontSize}
-          ><i class="icon-f">text_decrease</i></button
+        <button class="btn hover:bg-primary-hover-token p-1" on:click={decreaseFontSize}>
+          <Icon iconName="text_decrease" /></button
```
```suggestion
          <Icon name="text_decrease" /></button
```

pretzL commented on packages/lms/src/lib/components/header/Settings.svelte:
```diff
@@ -141,14 +142,14 @@
         >
         <button class="btn hover:bg-primary-hover-token p-1" on:click={resetFontSize}>
-          <i class="icon-f">refresh</i></button
+          <Icon iconName="refresh" /></button
```
```suggestion
          <Icon name="refresh" /></button
```

pretzL commented on packages/lms/src/lib/components/navigation/Entity.svelte:
```diff
@@ -1,74 +1,75 @@
+  <div class="entity-inner">
+    <div class="entity-header hover:bg-primary-hover-token rounded-container-token p-2">
+      <a href={entity.browserPath} on:click={drawerClose} class="entity-title flex flex-row items-center">
+        <Icon iconName={icon} />
```
```suggestion
        <Icon name={icon} />
```

pretzL commented on packages/lms/src/lib/components/navigation/Entity.svelte:
```diff
@@ -1,74 +1,75 @@
+      </a>
+      {#if entity.children.length}
+        <button on:click={toggle} class="btn hover:bg-primary-hover-token p-0">
+          <Icon iconName={open ? 'expand_less' : 'expand_more'} />
```
```suggestion
          <Icon name={open ? 'expand_less' : 'expand_more'} />
```

pretzL commented on packages/lms/src/lib/components/navigation/icons/AddRemoveBookmarks.svelte:
```diff
@@ -37,12 +38,12 @@
   >
     {#if hasBookmarks}
-      <div class="app-rail-tile-icon"><i class="icon-f">bookmark_remove</i></div>
+      <Icon iconName="bookmark_remove" />
```
```suggestion
      <Icon name="bookmark_remove" />
```

pretzL commented on packages/lms/src/lib/components/navigation/icons/AddRemoveBookmarks.svelte:
```diff
@@ -37,12 +38,12 @@
+      <Icon iconName="bookmark_remove" />
     {:else}
-      <div class="app-rail-tile-icon"><i class="icon-f">bookmark_add</i></div>
+      <Icon iconName="bookmark_add" />
```
```suggestion
      <Icon name="bookmark_add" />
```

pretzL commented on packages/lms/src/lib/components/navigation/icons/Bookmarks.svelte:
```diff
@@ -22,7 +23,7 @@
     on:click={() => modalStore.trigger(searchModal)}
   >
-    <div class="app-rail-tile-icon"><i class="icon-f">bookmark</i></div>
+    <Icon iconName="bookmark" />
```
```suggestion
    <Icon name="bookmark" />
```

pretzL commented on packages/lms/src/lib/components/reader/Reader.svelte:
```diff
@@ -124,7 +125,7 @@
     <div>
       <button class="btn hover:bg-primary-hover-token" use:popup={audioSettings}>
-        <i class="icon-f">settings</i></button
+        <Icon iconName="settings" /></button
```
```suggestion
        <Icon name="settings" /></button
```

pretzL commented on packages/lms/src/lib/components/ui/Icon.svelte:
```diff
@@ -0,0 +1,6 @@
+  import 'material-icons/iconfont/material-icons.css';
+</script>
+
+<span class="material-icons">{iconName}</span>
```
```suggestion
<span class="material-icons" {...$restProps}>{iconName}</span>
```
I'm a huge fan of the `$restProps` feature Svelte offers. If you suddenly need some extra styling applied directly to the `span` you can then use the in-line `style` attribute, for example. Or if you need to add eventListeners you can do that directly without having to edit the Icon component.

pretzL commented on packages/lms/src/lib/components/ui/Icon.svelte:
```diff
@@ -0,0 +1,6 @@
+<script lang="ts">
+  export let iconName: string;
```
```suggestion
  export let name: string;
```

pretzL commented on packages/lms/src/lib/styles/_components.scss:
```diff
@@ -14,11 +14,11 @@
 
-.icon-f {
+.material-icons {
   @include icon;
 }
 
-.icon-o {
+.material-icons {
   @include icon;
   font-variation-settings: 'FILL' 0;
 }
```
This isn't really DRY compliant. Also, shouldn't these perhaps be variance options as props in the component? Like you can then use the component like so `<Icon name="home" variant="filled" />` and it'd give you a filled icon?

Fermain replied:
That's a nice idea but might be a good idea for a new task

pretzL reviewed (changes requested):
The `iconName` versus `name` is debatable, but I personally don't see a reason to use an unnecessarily long prop name for no reason.
Rest should probably be fixed.

Christonn93 commented:
I'll do the changes today and push it, Thanks for the feedback.

pretzL commented:
> I'll do the changes today and push it, Thanks for the feedback.

Just in case you don't know, go to the "files" tab and click `Add suggestion to batch` on each of the suggestions you want to commit, then commit. Saves you a lot of time

Fermain commented on packages/lms/src/lib/components/header/Search.svelte:
```diff
@@ -154,7 +155,8 @@
       class="btn hover:bg-primary-hover-token sm:variant-filled-primary sm:rounded-l-none sm:hover:bg-primary-active-token"
-      ><i class="icon-f">search</i></button
+    >
+      <Icon iconName="search" /></button
```
```suggestion
      <Icon name="search" /></button
```

pretzL reviewed (approved):
Lgtm
